Feat/datepicker month navigation

rymeskar opened the pull request:
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Fully mimic Fabric behavior:
* Do not change focus upon enter-navigating month.
* Do not change the grid immediately the month changes but only once the date stops being visible.
* Memoize the dayGrid for better performance.

#### Focus areas to test

(optional)

Commit:
Do not focus on month change

Commit:
Adding memoization

Commit:
Mimic behavior of Fabric datepicker

Commit:
Initial formatting string

Commit:
Properly formatting aria label for calendar cell

Commit:
Merged

Commit:
Smooth behavior when focusing

Commit:
Merge branch 'feat/adding_more_datepicker_examples' into feat/datepicker_month_navigation

Commit:
Fixing tests

Commit:
Updating localization strings

pompomon commented on packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx:
```diff
@@ -143,15 +147,36 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
+  );
+
+  const normalizeDateInGrid = (date: Date): Date => {
+    const result = new Date(date.getTime());
```
```suggestion
    const result = new Date(date);
```

Would this give the same result?

pompomon commented on packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx:
```diff
@@ -143,15 +147,36 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
+  };
+
+  const getRidOfSecondsMinutesHours = (date: Date): Date => {
+    const result = new Date(date.getTime());
```
```suggestion
    const result = new Date(date.getFullYear(), date.getMonth(), date.getDate());
```
Like in https://github.com/microsoft/fluentui/blob/5fbb948524f5c80cf59378e57ac898fabd6f3273/packages/date-time-utilities/src/dateGrid/getDayGrid.ts#L37?

So we don't need to reset minutes/seconds/etc

pompomon commented on packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx:
```diff
@@ -143,15 +147,36 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
-  // TODO: make new date instances, so they are less mutable
-  const [gridNavigatedDate, setGridNavigatedDate] = React.useState<Date>(navigatedDate || today || new Date());
+  const [gridNavigatedDate, setGridNavigatedDate] = React.useState<Date>(
+    () => new Date((navigatedDate || today || new Date()).getTime()),
```
```suggestion
    () => new Date((navigatedDate || today || new Date())),
```
Please would it be the same without `getTiime()`?

RajeshGoriga committed:
Added multi selection of legends for the line chart.. (#14758)

* added multiselection of legends for the line chart.. (initial commit)

* FIx tslint errors

* Change files

* Change files

Co-authored-by: v-sivsar <v-sivsar@microsoft.com>
Co-authored-by: Rajesh Goriga <v-gorraj@microsoft.com>

elisabethcvs committed:
use forwardRef for the UnifiedPicker (#14787)

* First attempt, is broken

* Forwarding works, fix lint errors

* Change files

* use component ref insstead

* Fix clearInput example build failure

xugao committed:
react-slider: use merge-styles instead of scss and move scss code to next folder  (#14786)

* update from scss to merge-styles

* update snaps

* Change files

layershifter committed:
chore: disable test, add iterations to existing (#14799)

* chore: disable test, add iterations to existing

* reduce iterations

ecraig12345 committed:
Add missing no-console eslint disable

vyhnalekl committed:
Fixed incorrect margins after d&d (UI builder) (#14384)

ecraig12345 committed:
Remove tslint references from react-avatar (#14807)

xugao committed:
react-toggle: remove usage of compose and scss, move latest code to next folder  (#14794)

* prep toggle for v8

* add next example

* update snapshots

* remove themeprovider from vrtest

* Change files

* fix build

xugao committed:
react-link: remove usage of compose and scss, move latest code to next folder (#14791)

* remove compose and scss from Link

* fix next

* update screener tests

* Change files

* update snapshots

* remove useLink export

sopranopillow committed:
React conformance: Updating doctrine and modifying merge function to use lodash. (#14344)

Co-authored-by: Elizabeth Craig <elcraig@microsoft.com>

Commit:
Fixing

Commit:
Merge branch 'feat/adding_more_datepicker_examples' into feat/datepicker_month_navigation

pompomon committed:
docs(Datepicker): Add custom header example (#14658)

Add custom header example with arrows on the sides and the label in the center
![image](https://user-images.githubusercontent.com/2802155/90879864-d91fb980-e3a7-11ea-8054-2c2b44741164.png)

rymeskar committed:
Feat/adding more datepicker examples (#14656)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Adding more datepicker examples.

#### Focus areas to test

(optional)

Commit:
Merge remote-tracking branch 'upstream/master' into feat/datepicker_month_navigation

Commit:
API file

Commit:
Change files

Commit:
Formatted string

Commit:
Updating package

Commit:
Updating props in DatepickerCalendarHeader

rymeskar committed:
Update @fluentui-date-time-utilities-2020-08-31-11-26-38-feat-datepicker_month_navigation.json

rymeskar committed:
Update @fluentui-date-time-utilities-2020-08-31-11-26-38-feat-datepicker_month_navigation.json
